docs/source/tutorial_optimization.md

```diff
@@ -584,215 +479,146 @@ reference_mu = opt_fom_result.x
 
-opt_dict = {'opt_rom': opt_rom}
-opt_along_path_adaptively_result = optimize(enrich_adaptively_and_compute_objective_function,
-                                            opt_along_path_adaptively_minimization_data, ranges,
-                                            gradient=partial(compute_gradient_with_opt_rom, opt_dict),
-                                            adaptive_enrichment=True, opt_dict=opt_dict)
-```
+tic = perf_counter()
+tr_mu, tr_minimization_data = coercive_rb_trust_region(pdeopt_reductor, parameter_space=parameter_space,
+                                                       initial_guess=np.array(initial_guess))
+toc = perf_counter()
 
-```{code-cell}
-report(opt_along_path_adaptively_result, opt_along_path_adaptively_minimization_data, reference_mu)
+tr_minimization_data['time'] = toc - tic
+tr_output = fom.output(tr_mu)[0, 0]
 ```
 
-Now, we actually only needed {math}`4` enrichments and ended up with an
-approximation error of about `1e-07` while getting the highest speed up
-amongst all methods that we have seen above. Note, however, that this is
-still dependent on the tolerance `atol=1e-2` that we chose without
-knowing that this tolerance suffices to reach the actual minimum.
-An easy way around this would be to do one optimization step with the FOM
-after converging. If this changes anything, the ROM tolerance `atol`
-was too large. To conclude, we once again
-compare all methods that we have discussed in this notebook.
+Now, we only need {math}`4` enrichments and end up with an approximation
+error of about `1e-06` which is comparable to the result obtained from the
+previous methods. Unfortunately, the trust-region method takes longer because of
+the overhead of ROM evaluations and line searches performed in the BFGS step.
+To conclude, we once again compare all methods that we have discussed in this notebook.
 
 ```{code-cell}
+from pymordemos.trust_region import report as tr_report
+reference_fun = fom_objective_functional(reference_mu)
+
 print('FOM with finite differences')
 report(fom_result, reference_minimization_data, reference_mu)
 
-print('\nROM with finite differences')
-report(rom_result, RB_minimization_data, reference_mu)
-
 print('\nFOM with gradient')
 report(opt_fom_result, opt_fom_minimization_data, reference_mu)
 
-print('\nROM with gradient')
-report(opt_rom_result, opt_rom_minimization_data, reference_mu)
-
-print('\nAlways enrich along the path')
-report(opt_along_path_result, opt_along_path_minimization_data, reference_mu)
-
-print('\nAdaptively enrich along the path')
-report(opt_along_path_adaptively_result, opt_along_path_adaptively_minimization_data, reference_mu)
+tr_report(tr_mu, tr_output, reference_mu, reference_fun, tr_minimization_data,
+    parameter_space.parameters.parse, descriptor=' of optimization with adaptive ROM model and TR method')
 ```
 
+It is apparent that for this example no drastic changes occur when using the different
```

Comment:
So we don't need real gradients in the first place? When can we expect to need them?